packages/next/lib/helpers/get-version-info.ts

```diff
@@ -0,0 +1,39 @@
+import type { Readable } from 'stream'
+
+export async function getVersionInfo(): Promise<any> {
+  try {
+    const response = await fetch('https://registry.npmjs.org/next', {
+      headers: {
+        accept:
+          'application/vnd.npm.install-v1+json; q=1.0, application/json; q=0.8, */*',
+      },
+    })
```

Comment:
npm registry API contains way too much information, and I actually have a good idea:

- https://unpkg.com/next@latest/package.json
- https://unpkg.com/next@canary/package.json

unpkg will always 302 redirect the dist tag to the corresponding files (with the latest satisfied version).